### ISSUE TITLE
Bug: Hidden curves should not count to "Same Color" restrictions

### ISSUE THREAD

**Issue by mducle** [mantidproject/mslice#548]
### Expected behavior

Can change the color of a line to another color which does not appear on the plot.

### Actual behavior

There is a check which ensures that lines on the plot do not have the same color. This check does not take into account whether a curve is hidden or not. So if the curve is hidden, and the user wants an unhidden curve to have the same color, this still gives an error.

### Steps to reproduce the behavior

Plot multiple (line-plot) curves using the `Cut` tab. Click on the wheel (settings icon) and hide one of the curves (unset one of the `Show Lines` checkbox). Set another curve to the same color as the hidden curve -> an error dialog will pop up saying that you cannot have two lines of the same color.

### Platforms affected

All.

**Comment by SilkeSchomann** [comment]
@mducle Is this still a problem? The hidden curve could just get deleted.

**Comment by mducle** [comment]
@SilkeSchomann yes, it's still an issue with Mantid nightly 6.1.20210802.1714 - also I found another issue - the settings are not reset when the plot is redone (e.g. plot a series of cuts [overplot], hide one of them, and replot the same number of curves, click the settings wheel icon and you'll see the previously hidden curve will show as being hidden even though it is not).
